commit 2f5581e82fe46e0f2bc68d0c69176f3338fda4b0
Author: Keith Busch <keith.busch@intel.com>
Date:   Tue Jan 8 09:37:43 2019 -0700

    nvme: pad fake subsys NQN vid and ssvid with zeros
    
    [ Upstream commit 3da584f57133e51aeb84aaefae5e3d69531a1e4f ]
    
    We need to preserve the leading zeros in the vid and ssvid when generating
    a unique NQN. Truncating these may lead to naming collisions.
    
    Signed-off-by: Keith Busch <keith.busch@intel.com>
    Signed-off-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/nvme/host/core.c b/drivers/nvme/host/core.c
index e5bddae16ed4..e0d2b7473901 100644
--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -2095,7 +2095,7 @@ static void nvme_init_subnqn(struct nvme_subsystem *subsys, struct nvme_ctrl *ct
 
 	/* Generate a "fake" NQN per Figure 254 in NVMe 1.3 + ECN 001 */
 	off = snprintf(subsys->subnqn, NVMF_NQN_SIZE,
-			"nqn.2014.08.org.nvmexpress:%4x%4x",
+			"nqn.2014.08.org.nvmexpress:%04x%04x",
 			le16_to_cpu(id->vid), le16_to_cpu(id->ssvid));
 	memcpy(subsys->subnqn + off, id->sn, sizeof(id->sn));
 	off += sizeof(id->sn);
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1391_2f5581e82fe46e0f2bc68d0c69176f3338fda4b0.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1391_2f5581e82fe46e0f2bc68d0c69176f3338fda4b0.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
